tests/python/DodgyBearTest.py

```diff
@@ -0,0 +1,102 @@
+                         '<<<<<<< HEAD'
+                         '\nvar = "test"\n=======\nvar = "test"\n'
+                         '>>>>>>> branch')
+        self.assertEqual(self.get_results(), [
+                         'Possible SCM diff in code',
+                         'Possible SCM diff in code'])
+        os.remove('diff.py')
+
+    def test_password(self):
+        create_test_file('passwords.py',
+                         'FACEBOOK_PASSWORD = "random@123"\n'
+                         'PASSWORD_TO_GITHUB = "random@123"\n'
+                         'PASSWORD = "random@123"\n'
+                         'PASSWORD = ""')
+        self.assertEqual(self.get_results(), [
+                         'Possible hardcoded password',
+                         'Possible hardcoded password',
+                         'Possible hardcoded password'])
+        os.remove('passwords.py')
+
+    def test_secret(self):
+        create_test_file('secrets.py',
+                         'SECRET = "random@123"\n'
+                         'SECRET_KEY = "random@123"\n'
+                         'THE_SUPER_SECRET = "random@123"')
+        self.assertEqual(self.get_results(), [
+                         'Possible hardcoded secret key',
+                         'Possible hardcoded secret key',
+                         'Possible hardcoded secret key'])
+        os.remove('secrets.py')
+
+    def test_ssh_private_key(self):
+        create_test_file('ssh_private_key.py',
+                         '-----BEGIN RSA PRIVATE KEY-----\n'
+                         'RANDOM@123\n'
+                         '-----END RSA PRIVATE KEY-----')
+        self.assertEqual(self.get_results(), [
+                         'Possible SSH private key',
+                         'Possible SSH private key'])
+        os.remove('ssh_private_key.py')
+
+    def test_ssh_public_key(self):
+        create_test_file('ssh_public_key.py',
+                         'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDMqZ2FO/Y'
+                         'y+ci0+9y3xUeonX/4tmfYPiAkjsfUWWkykHkCYUlTVS92pr'
+                         'aiTO256xy29osYk1xs7U0OpLztfHanHBuUhZz0pn/RvyYGe'
+                         'q3i5H3Ne/Z21TldMakhaKMB42V1C2+OV0xnjoEtPYUVG8ici'
+                         '7EzjLyDotRkYigRouI26PsmKPyyfqZ3TYVfMtkP21RdcI4/l'
+                         'na6d1dBmrQ3ly0B6g+pEf8ObaBriu0Fest0yZhyF8hsgueGu'
+                         'gZFXzjaAu68ib99nP17w111cbej9jUNnyfseykl5ngqrvsP3'
+                         'mGKw739z+p0vIIchHYa3jn6xW8fim/PH4gtcWC8mUXlv6XT '
+                         'random@123')
+        self.assertEqual(self.get_results(), ['Possible SSH public key'])
+        os.remove('ssh_public_key.py')
+
+    def tearDown(self):
+        current = os.listdir()
+        test_files = ['amazon.py', 'diff.py', 'passwords.py',
+                      'secrets.py', 'ssh_private_key.py', 'ssh_public_key.py']
+        for f in test_files:
+            if f in current:
+                os.remove(f)
```

Comment:
yeah properly using test files makes this unnecessary^^

Reply:
(Btw you should always remove temp files where you created them in a test function, best is using context managers for this)

Reply:
@Makman2 We did discuss this before.
The problem was that dodgy only runs for files in the current directory. No files can be specified. It doesn't accept absolute/relative paths to a file which should be checked.

So, we cannot use test files here. But rather create test files in the pwd, test dodgy and then remove the created files.

:(

Reply:
Ahhh I remember now... Yeah sorry it's been a while. Can you maybe execute coala / the bear inside a different working directory? Maybe patching some variables?

Reply:
Ya, we have a decorator which changes directory.

Reply:
I've emailed the maintainer to ask them whether we can implement https://github.com/landscapeio/dodgy/issues/5 for them.

Reply:
https://github.com/zricethezav/gitleaks

Reply:
does `gitleaks` support individual files?  if not, it is still not ideal.  Create a new issue for gitleaks if you like.

Anyway, dodgy maintainer replied and said they are happy to review PRs for landscapeio/dodgy#5 .

That gets it usable, and then this PR can be finished.

Reply:
`gitleaks` doesn't support individual files, but does accept a path to a directory which `dodgy` isn't capable of. Although it works only if it's a path to a git repo, and unlike `dodgy`, it runs checks on the whole commit history.
Also, it is capable of running on remote repositories, and on all of the repositories of a particular user/organization at once.